### PR TITLE
Haichen userlogin/signup

### PR DESCRIPTION
**Pull request by xuh5**
image
image
image
unauthorized user will redirect to login if click on button "request event , donation, send photo"

### CODE REVIEW

File: src/pages/Auth.css [hunk 1]
```diff
@@ -0,0 +1,139 @@
+/* Page Container Styling */
+.page-container {
+    display: flex;
+    flex-direction: column; /* Stack elements vertically */
+    align-items: center;    /* Center content horizontally */
+    justify-content: flex-start; /* Start content from the top */
+    min-height: 100vh;      /* Take full viewport height */
+    background-color: #FAF5F2; /* Background color */
+    padding-top: 300px;     /* Add space at the top */
+}
```
Comment (Herbert540) [inline src/pages/Auth.css:10]: This affects other pages, can you update that `padding-top: 300px;`

File: src/pages/UserLogin.jsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            await addUserData(
+              {
+                username: user.displayName|| 'Google User',
+                email: user.email,
+                provider: userCredential.providerId || 'google',
+              },
+              user.uid
+            );
+          }
+
```
Comment (Herbert540) [inline src/pages/UserLogin.jsx:40]: This is some good stuff!

**Review by Herbert540 — APPROVED** [review]
lgtm, the only thing is the `page-container` styling

File: src/pages/Auth.css (hunk 1 above)
Comment (xuh5) [reply src/pages/Auth.css:10]: its just because the inbox page shares same class naming. updated to page-container-auth-page

**Pull request merged**